teuthology/orchestra/remote.py

```diff
@@ -461,6 +461,30 @@ def _set_iface_and_cidr(self):
                 return
         raise RuntimeError("Could not determine interface/CIDR!")
 
+
+    def resolve_ip(self, name=None, ipv='4') -> str:
+        """
+        Resolve IP address of the remote host via remote host
+
+        Because remote object maybe behind bastion host we need
+        the remote host address resolvable from remote side.
+        So in order to the ip address we just call `host` remotely
+        and parse output like:
+            'smithi001.front.sepia.ceph.com has address 172.21.15.1\n'
+        """
+        hostname = name or self.hostname
+        if str(ipv) in ['4', '6']:
+            remote_host_ip = self.sh(f'host -{ipv} {hostname}')
+        else:
+            raise Exception(f'Unknown IP version {ipv}, expected 4 or 6')
+        if 'has address' in remote_host_ip:
+            (host, ip) = remote_host_ip.strip().split(' has address ')
```

Comment:
If the name resolves to multiple addresses, we will get:
```
ValueError: too many values to unpack (expected 2)
```

Reply:
hm, didn't test it against google.com, it has indeed strange response:
```
$ host -4 google.com
google.com has address 142.250.185.110
google.com has IPv6 address 2a00:1450:4001:80f::200e
google.com mail is handled by 10 smtp.google.com.
```
I only verified this against target nodes. I guess in this case it is only needed to find first occurrence of "has address" line if any.